doc/md/fundamentals/3-declarations/2-function-declarations.md

```diff
@@ -0,0 +1,127 @@
+---
+sidebar_position: 2
+---
+
+# Functions declarations
+
+A function in Motoko is a reusable block of code that accepts inputs, performs computations or actions, and optionally returns a result. Functions can be either named or anonymous and may explicitly define the types of their parameters and return values for clarity and type safety.
+
+## Declaring a function
+
+Functions are declared using the `func` keyword. A named function assigns the function to an identifier, making it accessible for reuse and enabling recursion. This allows the function to refer to itself by name within its own body. In contrast, anonymous functions (e.g., lambda functions or function expressions) do not have a name and are typically defined inline. These functions are often assigned to variables or passed directly as arguments to other functions.
+
+```motoko no-repl
+// The function is named 'add'
+// 'a : Int' and 'b : Int' are parameters with types.
+// 'Int' is the return type.
+func add(a : Int, b : Int) : Int {
+    // Exits the function and provides a result.
+    return a + b;
+}
+```
+
+## Calling a function
+
+To execute a function, you simply call it by its name (or reference, in the case of anonymous functions) and pass the required arguments in parentheses. The arguments must match the number and types specified in the function’s definition.
+
+```motoko no-repl
+func add(a : Int, b : Int) : Int {
+    return a + b;
+};
+add(3, 5);
+```
+
+## Functions as values (anonymous functions)
+
+Functions in Motoko are first-class values, meaning they can be treated like any other value in the language. They can be assigned to variables, stored in data structures such as arrays or records, and passed as arguments to other functions.
+
+```motoko no-repl
+let double = func (x : Int) : Int { x * 2 };
+let applyFunction = func (f : (Int) -> Int, value : Int) : Int { f(value) };
+
+applyFunction(double, 10);
+```
+
+`applyFunction` takes a function `f` and applies it to `value`.  `double` is passed as an parameter, allowing to construct higher-order functions.
```

Comment:
```suggestion
`applyFunction` takes a function `f` and applies it to `value`. Because `applyFunction` takes a function as a parameter, it it is a _higher-order_ function.

In the call, the function `double` is passed as an argument.
```